prometheus-metrics-exporter-httpserver/src/main/java/io/prometheus/metrics/exporter/httpserver/HTTPServer.java

```diff
@@ -55,7 +55,13 @@ private HTTPServer(PrometheusProperties config, ExecutorService executorService,
         registerHandler("/", defaultHandler == null ? new DefaultHandler() : defaultHandler, authenticator);
         registerHandler("/metrics", new MetricsHandler(config, registry), authenticator);
         registerHandler("/-/healthy", new HealthyHandler(), authenticator);
-        this.server.start();
+        try {
+            executorService.submit(() -> this.server.start()).get();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e.getCause());
```

Comment:
When wrapping exceptions, pass the exception being wrapped.

```
throw new RuntimeException(e);
```

Reply:
OK